### PR TITLE
Added fix for issue #25332

### PR DESCRIPTION
**Pull request by yarwest**
Made sure that the newly entered password is not the same as the
existing password. This is done using a simple if statement comparing
already defined variables for the newly entered and the old password. If
the variables are the same a fitting error will be passed.

(used my girlfriend's account on accident, that explains the reverts)

### CODE REVIEW

**Comment by DeepDiver1975** [comment]
Please Squash your commits. Thx

File: settings/ChangePassword/Controller.php [hunk 1]
```diff
@@ -45,6 +45,11 @@ public static function changePersonalPassword($args) {
 			\OC_JSON::error(array("data" => array("message" => $l->t("Wrong password")) ));
 			exit();
 		}
+        if ($oldPassword == $password) {
```
Comment (DeepDiver1975) [inline settings/ChangePassword/Controller.php]: Indent looks off

Comment (DeepDiver1975) [inline settings/ChangePassword/Controller.php]: Should be ===

**Comment by yarwest** [comment]
I have sent the contributor agreement yesterday, but I hereby declare that my contributions to this project are MIT licensed.

File: settings/ChangePassword/Controller.php (hunk 1 above)
Comment (yarwest) [inline settings/ChangePassword/Controller.php]: The indents are fine within the IDE and the GitHub desktop app, for some reason they are off on the webplatform.

**Comment by yarwest** [comment]
As you might notice, I am fairly new with github so I am sorry if my commits are a mess.

**Comment by PVince81** [comment]
Ok, we can squash on merge then

File: settings/ChangePassword/Controller.php [hunk 2]
```diff
@@ -45,6 +45,11 @@ public static function changePersonalPassword($args) {
 			\OC_JSON::error(array("data" => array("message" => $l->t("Wrong password")) ));
 			exit();
 		}
+        if ($oldPassword === $password) {
+            $l = new \OC_L10n('settings');
+			\OC_JSON::error(array("data" => array("message" => $l->t("The new password can not be the same as the previous one")) ));
+			exit();
+        }
```
Comment (PVince81) [inline settings/ChangePassword/Controller.php]: indentation is wrong, please use tabs

**Comment by DeepDiver1975** [comment]
Contribute agreement has received. Thx

File: settings/ChangePassword/Controller.php [hunk 3]
```diff
@@ -45,6 +45,11 @@ public static function changePersonalPassword($args) {
 			\OC_JSON::error(array("data" => array("message" => $l->t("Wrong password")) ));
 			exit();
 		}
+	        if ($oldPassword === $password) {
+	            $l = new \OC_L10n('settings');
+				\OC_JSON::error(array("data" => array("message" => $l->t("The new password can not be the same as the previous one")) ));
+				exit();
```
Comment (PVince81) [inline settings/ChangePassword/Controller.php]: The tabs still look off. What editor are you using ?

**Comment by PVince81** [comment]
Tested, works.

However for some reason the error message's position seems to have some trouble:
password-error

If you don't mind, could you set the CSS classes "msg" and "error" on the "password-error" element ? It will make it red and more flashy.

**Comment by yarwest** [comment]
For the initial commit I have used Brackets (in which the tabs were completely fine) after that I have repeatedly made changes in both Brackets and GitHub's online editor. I have no clue why the tabs keep freaking out.

**Comment by PVince81** [comment]
Looks good now :+1: 

I ran the JS unit tests locally and they still pass.

**Comment by PVince81** [comment]
Squashed + merged

**Pull request merged**